src/topology/algebra/order/left_right_lim.lean

```diff
@@ -0,0 +1,277 @@
+  calc right_lim f x ≤ f a : hf.right_lim_le xa
+  ... ≤ left_lim f y : hf.le_left_lim ay
+end
+
+variables [topological_space α] [order_topology α]
+
+lemma tendsto_left_lim (x : α) : tendsto f (𝓝[<] x) (𝓝 (left_lim f x)) :=
+begin
+  rcases eq_or_ne (𝓝[<] x) ⊥ with h'|h',
+  { simp [h'] },
+  rw left_lim_eq_Sup hf h',
+  exact hf.tendsto_nhds_within_Iio x
+end
+
+lemma tendsto_right_lim (x : α) :
+  tendsto f (𝓝[>] x) (𝓝 (right_lim f x)) :=
+@monotone.tendsto_left_lim αᵒᵈ βᵒᵈ _ _ _ _ f hf.dual _ _ x
+
+/-- A monotone function is continuous to the left at a point if and only if its left limit
+coincides with the value of the function. -/
+lemma continuous_within_at_Iio_iff_left_lim_eq  :
+  continuous_within_at f (Iio x) x ↔ left_lim f x = f x :=
+begin
+  rcases eq_or_ne (𝓝[<] x) ⊥ with h'|h',
+  { simp [left_lim_eq_of_eq_bot f h', continuous_within_at, h'] },
+  haveI : (𝓝[Iio x] x).ne_bot := ne_bot_iff.2 h',
+  refine ⟨λ h, tendsto_nhds_unique (hf.tendsto_left_lim x) h.tendsto, λ h, _⟩,
+  have := hf.tendsto_left_lim x,
+  rwa h at this,
+end
+
+/-- A monotone function is continuous to the right at a point if and only if its right limit
+coincides with the value of the function. -/
+lemma continuous_within_at_Ioi_iff_right_lim_eq :
+  continuous_within_at f (Ioi x) x ↔ right_lim f x = f x :=
+@continuous_within_at_Iio_iff_left_lim_eq αᵒᵈ βᵒᵈ _ _ _ _ f hf.dual x _ _
+
+/-- A monotone function is continuous at a point if and only if its left and right limits
+coincide. -/
+lemma continuous_at_iff_left_lim_eq_right_lim :
+  continuous_at f x ↔ left_lim f x = right_lim f x :=
+begin
+  refine ⟨λ h, _, λ h, _⟩,
+  { have A : left_lim f x = f x,
+      from (hf.continuous_within_at_Iio_iff_left_lim_eq).1 h.continuous_within_at,
+    have B : right_lim f x = f x,
+      from (hf.continuous_within_at_Ioi_iff_right_lim_eq).1 h.continuous_within_at,
+    exact A.trans B.symm },
+  { have h' : left_lim f x = f x,
+    { apply le_antisymm (left_lim_le hf (le_refl _)),
+      rw h,
+      exact le_right_lim hf (le_refl _) },
+    refine continuous_at_iff_continuous_left'_right'.2 ⟨_, _⟩,
+    { exact hf.continuous_within_at_Iio_iff_left_lim_eq.2 h' },
+    { rw h at h',
+      exact hf.continuous_within_at_Ioi_iff_right_lim_eq.2 h' } },
+end
+
+open function
+
+/-- In a second countable space, the set of points where a monotone function is not right-continuous
+is at most countable. Superseded by `countable_not_continuous_at` which gives the two-sided
+version. -/
+lemma countable_not_continuous_within_at_Ioi [topological_space.second_countable_topology β] :
+  set.countable {x | ¬(continuous_within_at f (Ioi x) x)} :=
+begin
+  /- If `f` is not continuous on the right at `x`, there is an inverval `(f x, z x)` which is not
+  reached by `f`. This gives a family of disjoint open intervals in `β`. Such a family can only
+  be countable as `β` is second-countable. -/
+  nontriviality α,
+  inhabit α,
+  haveI : nonempty β := ⟨f default⟩,
```

Comment:
`nonempty.map` might golf this line and the one above slightly

Reply:
In fact, this is not even needed as Lean knows that a type with a Sup is nonempty!